"response.body?.pipeTo is not a function" while downloading filter lists using Node.js v16

@hlqviet Tagging you because this might be related to #25 (though I'll try to fix this on my own in my free time).

This is most likely caused by the old Node.js version, that we have to support because that's what the original GitHub Actions workflow uses.

**Environment**

```
  node: v16.20.2
  npm: 8.19.4
  yarn: 1.22.19
```

**Error**

```
> Run bash ./get_recommended_whitelist.sh
file:///home/runner/work/***/***/cloudflare-gateway-pihole-scripts/lib/utils.js:69
    await response.body?.pipeTo(writableStream, { preventClose: true });
                         ^

TypeError: response.body?.pipeTo is not a function
    at downloadFiles (file:///home/runner/work/***/***/cloudflare-gateway-pihole-scripts/lib/utils.js:69:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async downloadLists (file:///home/runner/work/***/***/cloudflare-gateway-pihole-scripts/download_lists.js:18:3)
    at async file:///home/runner/work/***/***/cloudflare-gateway-pihole-scripts/download_lists.js:32:5
Error: Process completed with exit code 1.
```

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -1,8 +1,7 @@
 import { once } from "events";
-import { createReadStream, createWriteStream } from "fs";
+import { ReadStream, createReadStream, createWriteStream } from "fs";
 import { basename } from "path";
 import { createInterface } from "readline";
-import { WritableStream } from "stream/web";
 
 if (!globalThis.fetch) {
   globalThis.fetch = (await import("node-fetch")).default;
@@ -58,19 +57,19 @@ export const isComment = (value) =>
  */
 export const downloadFiles = async (filePath, urls) => {
   const writeStream = createWriteStream(filePath, { flags: "a" });
-  const writableStream = new WritableStream({
-    write(chunk) {
-      writeStream.write(chunk);
-    },
-  });
   const responses = await Promise.all(urls.map((url) => fetch(url)));
 
   for (const response of responses) {
-    await response.body?.pipeTo(writableStream, { preventClose: true });
-    writeStream.write("\n");
-  }
+    const readable = ReadStream.from(response.body, {
+      autoDestroy: true,
+    });
 
-  await writableStream.close();
+    readable.on("end", () => {
+      writeStream.write("\n");
+    });
+
+    readable.pipe(writeStream, { end: false });
+  }
 };
 
 /**

